src/test/java/com/github/commonsrdf/api/AbstractRDFTermFactoryTest.java

```diff
@@ -0,0 +1,237 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.commonsrdf.api;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotSame;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public abstract class AbstractRDFTermFactoryTest {
+
+	private RDFTermFactory factory;
+
+	@Test
+	public void createBlankNode() throws Exception {
+		BlankNode bnode = factory.createBlankNode();
+		String ntriplesString = bnode.ntriplesString();
+		assertTrue("ntriples must start with _:",
+				ntriplesString.startsWith("_:"));
+		assertTrue("Internal identifier can't be empty", bnode
+				.internalIdentifier().length() > 0);
+		assertEquals("ntriples does not correspond with internal identifier",
```

Comment:
I'm not sure I'm with you. The result of `ntriplesString()` should surely be a valid ntriples string for a blank node? If it can't create one I have thrown an `IllegalStateException` - perhaps this could be hinted at in the interface of `RDFTerm.ntriplesString()`?

Does this mean that you can also have an `IRI` which can't give a valid `ntriplesString`?

Are you saying that if an implementation decides to support non-compatible bnode identifiers (say one containing `:`), then it is also free to use such in the auto-generated identifier of `factory.createBlankNode()` - and hence have default blank nodes which can't respond to `ntriplesString()` method? This sounds like bad practice and exactly something this unit test should detect.

If you really want to support that (which I would not like..) - then we should make the whole ntriplesString method optional with `throws UnsupportedOperationException` hint.